### PR TITLE
Pipeline, Anastasia Gubarenko - 21FPL2

### PR DESCRIPTION
**Pull request by cybern00dle**

### CODE REVIEW

File: lab_6_pipeline/pipeline.py [hunk 1]
```diff
@@ -18,21 +33,47 @@ def __init__(self, path_to_raw_txt_data: Path):
...
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+
+        if not any(self.path_to_raw_txt_data.iterdir()):
+            raise EmptyDirectoryError
+
+        raw_files = [text for text in self.path_to_raw_txt_data.glob(r'*_raw.txt')]
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py]: cast to list instead of comprehension

File: lab_6_pipeline/pipeline.py [hunk 2]
```diff
@@ -18,21 +33,47 @@ def __init__(self, path_to_raw_txt_data: Path):
...
+
+        list_existing = sorted(int(re.search(r'\d+', file.stem)[0]) for file in raw_files)
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py]: what if there is no number in naming?

File: lab_6_pipeline/pipeline.py [hunk 3]
```diff
@@ -18,21 +33,47 @@ def __init__(self, path_to_raw_txt_data: Path):
...
+        list_ideal = list(range(1, len(raw_files) + 1))
+        if list_existing != list_ideal:
+            raise InconsistentDatasetError
+
+        if not any(file.stat().st_size for file in raw_files):
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py]: strange condition. if not at least one file is not empy?

File: lab_6_pipeline/target_score.txt [hunk 4]
```diff
@@ -1 +1 @@
-0
+4
```
Comment (demid5111) [inline lab_6_pipeline/target_score.txt]: only 4?? 😢

**Review by demid5111 — CHANGES_REQUESTED** [review]
it is only a beginning of pipeline, but so little time left, hurry up!

File: lab_6_pipeline/pipeline.py [hunk 5]
```diff
@@ -18,21 +33,47 @@ def __init__(self, path_to_raw_txt_data: Path):
...
+            raise NotADirectoryError
+
+        if not any(self.path_to_raw_txt_data.iterdir()):
+            raise EmptyDirectoryError
+
+        raw_files = [text for text in self.path_to_raw_txt_data.glob(r'*_raw.txt')]
+
+        list_existing = sorted(int(re.search(r'\d+', file.stem)[0]) for file in raw_files)
+        list_ideal = list(range(1, len(raw_files) + 1))
+        if list_existing != list_ideal:
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py]: where is check for numbering of meta files?

**Review by demid5111 — CHANGES_REQUESTED** [review]
nothing new to review. hurry up, your PR is at risk of missing deadline

File: lab_6_pipeline/pipeline.py (hunk 5 above)
Comment (cybern00dle) [reply lab_6_pipeline/pipeline.py]: it wasn't needed for mark 4 but now it's done

File: lab_6_pipeline/pipeline.py (hunk 3 above)
Comment (cybern00dle) [reply lab_6_pipeline/pipeline.py]: fixed it

File: lab_6_pipeline/pipeline.py (hunk 2 above)
Comment (cybern00dle) [reply lab_6_pipeline/pipeline.py]: this cannot be my case as all files in this corpus are numbered

File: lab_6_pipeline/pipeline.py (hunk 1 above)
Comment (cybern00dle) [reply lab_6_pipeline/pipeline.py]: done, thank you

**Review by demid5111 — CHANGES_REQUESTED** [review]
# sent_id = 1
# text = Это позволит повысить финансовую устойчивость региональных и местных бюджетов и обеспечить более равномерное финансирование расходных обязательств.
1	.	.	PUNCT	_	_	0	root	_	_

**Pull request closed without merging**